source/solvers/postprocessing_cfd.cc

```diff
@@ -159,6 +159,151 @@ calculate_pressure_drop<3, TrilinosWrappers::MPI::BlockVector>(
+                {
+                  if (cell->face(face)->boundary_id() == inlet_boundary_id)
+                    {
+                      fe_face_values.reinit(cell, face);
+                      for (unsigned int q = 0; q < n_q_points; q++)
+                        {
+                          inlet_surface += fe_face_values.JxW(q);
+                          fe_face_values[velocities].get_function_values(
+                            evaluation_point, velocity_values);
+                          fe_face_values[pressure].get_function_values(
+                            evaluation_point, pressure_values);
+                          // Integration of density*(velocity^2 +
+                          // pressure), since the pressure in Lethe has units of
+                          // Length^2/Time^2
+                          momentum_inlet_boundary +=
+                            fe_face_values.JxW(q) *
+                            (velocity_values[q] * velocity_values[q] *
+                               fluid_density +
+                             pressure_values[q]);
+                        }
+                    }
+                  if (cell->face(face)->boundary_id() == outlet_boundary_id)
+                    {
+                      fe_face_values.reinit(cell, face);
+                      for (unsigned int q = 0; q < n_q_points; q++)
+                        {
+                          outlet_surface += fe_face_values.JxW(q);
+                          fe_face_values[velocities].get_function_values(
+                            evaluation_point, velocity_values);
+                          fe_face_values[pressure].get_function_values(
+                            evaluation_point, pressure_values);
+                          // Integration of density*(velocity^2 +
+                          // pressure), since the pressure in Lethe has units of
+                          // Length^2/Time^2
+                          momentum_outlet_boundary +=
+                            fe_face_values.JxW(q) *
+                            (velocity_values[q] * velocity_values[q] +
+                             pressure_values[q]) *
+                            fluid_density;
+                        }
+                    }
+                }
+            }
+        }
+    }
+  const MPI_Comm mpi_communicator = dof_handler.get_communicator();
+  momentum_inlet_boundary =
+    Utilities::MPI::sum(momentum_inlet_boundary, mpi_communicator);
+  inlet_surface = Utilities::MPI::sum(inlet_surface, mpi_communicator);
+  momentum_outlet_boundary =
+    Utilities::MPI::sum(momentum_outlet_boundary, mpi_communicator);
+  outlet_surface = Utilities::MPI::sum(outlet_surface, mpi_communicator);
+
+  momentum_outlet_boundary = momentum_outlet_boundary / outlet_surface;
+  momentum_inlet_boundary  = momentum_inlet_boundary / inlet_surface;
+  momentum_drop            = momentum_inlet_boundary - momentum_outlet_boundary;
```

Comment:
I disagree here. This is wrong. The momentum drop should not be divided by the area.... The momentum drop is momentum at outlet - momentum at inlet just straight up. This is not the same as when we calculate the pressure drop in which we want to know the difference between the average pressure at the outlet minus the average pressure at the inlet.